app/scripts/views/mixins/avatar-mixin.js

```diff
@@ -62,11 +62,10 @@ define([
 
     // Makes sure the account with uid has an uptodate image cache.
     // This should be called after fetching the current profile image.
-    _updateCachedProfileImage: function (uid, profileImage) {
-      var cachedAccount = this.user.getAccountByUid(uid);
-      if (! cachedAccount.isDefault()) {
-        cachedAccount.setProfileImage(profileImage);
-        this.user.setAccount(cachedAccount);
+    _updateCachedProfileImage: function (account, profileImage) {
+      if (! account.isDefault()) {
+        account.setProfileImage(profileImage);
+        this.user.setAccount(account);
```

Comment:
I've thought about a refactor where we instantiate accounts with a `user` that they delegate to with `account.save()`. It's worth opening an issue about that.

Saving automatically on changes seems a little magical for me– I'll need to think some more on it.